core/slim_sim.cpp

```diff
@@ -6201,6 +6202,53 @@ void SLiMSim::WritePopulationTable(tsk_table_collection_t *p_tables)
 	}
 }
 
+void SLiMSim::WriteTreeSequenceMetadata(tsk_table_collection_t *p_tables)
+{
+    int ret = 0;
+	nlohmann::json j;
+	j["SLiM"]["model_type"] = (ModelType() == SLiMModelType::kModelTypeWF) ? "WF" : "nonWF";
+	j["SLiM"]["generation"] = Generation();
+	j["SLiM"]["file_version"] = SLIM_TREES_FILE_VERSION_NEW;
+    if (spatial_dimensionality_ == 0) {
+        j["SLiM"]["spatial_dimensionality"] = "";
+    } else if (spatial_dimensionality_ == 1) {
+        j["SLiM"]["spatial_dimensionality"] = "x";
+    } else if (spatial_dimensionality_ == 2) {
+        j["SLiM"]["spatial_dimensionality"] = "xy";
+    } else {
+        j["SLiM"]["spatial_dimensionality"] = "xyz";
+    }
+    if (periodic_x_ & periodic_y_ & periodic_z_) {
+        j["SLiM"]["spatial_periodicity"] = "xyz";
+    } else if (periodic_x_ & periodic_y_) {
+        j["SLiM"]["spatial_periodicity"] = "xy";
+    } else if (periodic_x_ & periodic_z_) {
+        j["SLiM"]["spatial_periodicity"] = "xz";
+    } else if (periodic_y_ & periodic_z_) {
+        j["SLiM"]["spatial_periodicity"] = "yz";
+    } else if (periodic_x_) {
+        j["SLiM"]["spatial_periodicity"] = "x";
+    } else if (periodic_y_) {
+        j["SLiM"]["spatial_periodicity"] = "y";
+    } else {
+        j["SLiM"]["spatial_periodicity"] = "";
+    }
+	j["SLiM"]["separate_sexes"] = sex_enabled_ ? "true" : "false";
+	j["SLiM"]["nucleotide_based"] = nucleotide_based_ ? "true" : "false";
+	std::string metadata_str;
+    metadata_str = j.dump(4);
+
+    ret = tsk_table_collection_set_metadata(
+            p_tables, metadata_str.c_str(), (tsk_size_t)metadata_str.length());
+    if (ret != 0)
+        handle_error("tsk_table_collection_set_metadata", ret);
+    ret = tsk_table_collection_set_metadata_schema(
+            p_tables, SLIM_TSK_METADATA_SCHEMA.c_str(), SLIM_TSK_METADATA_SCHEMA.length());
+    if (ret != 0)
+        handle_error("tsk_table_collection_set_metadata_schema", ret);
+
+}
+
 void SLiMSim::WriteProvenanceTable(tsk_table_collection_t *p_tables, bool p_use_newlines, bool p_include_model)
 {
```

Comment:
some of the stuff we used to write to the provenance table should no longer be written there, right?  I was expecting to see diffs inside this function, removing that old cruft...?

Reply:
Well, we still want to record information about how the tree sequence was generated in provenance, so at least some of that stuff should stay. Why not all of it? I suppose that it's all in the script, if that gets embedded, but the script might not, so... I agree, it feels funny/redundant, but it's not redundant, since another script could load the tree sequence and write another provenance entry, and we got to have record that this happened.

Reply:
Oh, I meant, like, the generation number and the model type.  Those were in provenance but have now moved to ts metadata right?  So we don't need to write them to provenance any more?  But I'm not looking at the code right now, so maybe I'm being dumb.  :->

Reply:
No, you're right - right now, they are in both. But, generation number and model type both seem like good things to keep in provenance? Since: suppose that you (a) run a simulation and then (b) run another simulation that starts by reloading the output from (a). The provenance from both (a) and (b) will still be there, in the result, but the metadata will only contain (b). That's fine - the metadata tells us what we need to know to restart; while the provenance tells us how we got here. So - let's leave them in? Even though we won't be using provenance?

Reply:
Oh, but perhaps we don't need the "file_version" key in provenance?

Reply:
Well, it doesn't hurt, I suppose.

Reply:
Ah, I see.  Well, sure, that makes sense.  It's harmless, and somebody might someday thank us for leaving it in.